arbos/block_processor.go

```diff
@@ -455,6 +462,12 @@ func ProduceBlockAdvanced(
 		}
 	}
 
+	if sequencingHooks.BlockFilter != nil {
+		if err = sequencingHooks.BlockFilter(header, statedb, complete, receipts); err != nil {
+			return nil, nil, err
+		}
+	}
+
 	binary.BigEndian.PutUint64(header.Nonce[:], delayedMessagesRead)
 
 	FinalizeBlock(header, complete, statedb, chainConfig)
```

Comment:
Perhaps at the end of FinalizeBlock or at the end of this method, we need to error if `statedb.IsTxFiltered()`. Otherwise, the messages of the block will still be committed to consensus and published over the feed, as the ExecutionEngine calls `s.consensus.WriteMessageFromSequencer` before `appendBlock` (which itself would call Commit and error).

Reply:
good call! will fix it